main.go

```diff
@@ -80,6 +83,8 @@ func main() {
 	var replayBuf bytes.Buffer
 	tr := io.TeeReader(r, &replayBuf)
 
+	stopPulse := newPulse(*pulseIntervalPtr, ".")
```

Comment:
should we gate this feature by adding validation? e.g., check if we have `*pulseIntervalPtr` _and_ it is a valid `time.Duration` ?

I think this will prevent someone from entering an invalid value (for duration) and then wait n secs or minutes only to find out `exit status 2` .. in this case have `tparse` fail hard and early?

Reply:
There's a gate inside the constructor, because it makes the code easier to read. You don't have to deal with the potentially nil stopPulse. But I can move it outside if you think it will make it clearer.

Reply:
I was thinking `newPulse` would return an error, we call `os.Exit` and not even bother entering `parse.Process` ? 

But, the `go test` will still be running in the background .. so we're not actually gaining anything. Ideally we would grab the pid of the `go test` and kill it, but that's not worth it imo.

So nvm, I'm okay with this implementation. 

Let's just fix up https://github.com/mfridman/tparse/pull/36#discussion_r275022977 and merge 🎉